src/core/src/core_logic/PatchInstaller.py

```diff
@@ -240,7 +244,8 @@ def install_updates(self, maintenance_window, package_manager, simulate=False):
             if version == Constants.UA_ESM_REQUIRED:
```

Comment:
There is one more check for version == Constants.UA_ESM_REQUIRED in the function install_packages_in_batches. Similar changes are required there also.

Reply:
The parent packages which have this string will be skipped here itself. The Canonical team confirmed that no package which is non-esm will be dependent on package which is esm.

Reply:
The function install_packages_in_batches will be called before this point. This point is for sequential patching. We need to make similar changes inside "if version == Constants.UA_ESM_REQUIRED" in the function install_packages_in_batches also.

Reply:
Resolving this comment as UA_ESM_REQUIRED are already filtered and this if condition is removed as no longer required.